api/envoy/api/v2/core/base.proto

```diff
@@ -113,6 +142,21 @@ message Node {
   // version of Envoy is being tested in a heterogeneous fleet. This will be set
   // by Envoy in management server RPCs.
   string build_version = 5;
+
+  // Free-form string that identifies the entity requesting config.
+  // E.g. "envoy" or "grpc"
+  string user_agent_name = 6;
+
+  // Free-form string that identifies the version of the entity requesting config.
+  // E.g. "1.12.2" or "abcd1234", or "SpecialEnvoyBuild"
+  string user_agent_version = 7;
+
+  // Extension major version support
+  ExtensionVersions extension_versions = 8;
+
+  // Client feature support list. These are well known features described
+  // in the Envoy API repository for a given major version of an API.
+  repeated string client_features = 9;
```

Comment:
If we want this to be extensible, it should probably not be Envoy-specific.  Instead, it should probably use reverse DNS notation.

Reply:
+1

Reply:
Done.
@markdroth should we add feature for supporting Struct within Any only?

Reply:
Yes, that would be very useful.

Reply:
How should it be named? grpc.config.accept-struct-in-any-only?

Reply:
While gRPC will definitely use this option, I expect that other clients will too, since gRPC is not the only client that cannot handle `Any` fields with arbitrary embedded types.  As a result, it should probably have a name that does not start with `grpc.`.

How about something like `envoy.config.require-any-fields-contain-struct`?

Reply:
done